cloud/aws/elasticsearch/monitors-elasticsearch.tf

```diff
@@ -0,0 +1,106 @@
+resource "signalfx_detector" "heartbeat" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS ElasticSearch heartbeat"
+
+	program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('Nodes', filter=filter('namespace', 'AWS/ES') and filter('stat', 'mean') and ${module.filter-tags.filter_custom})
+		not_reporting.detector(stream=signal, resource_identifier=['DomainName'], duration='${var.heartbeat_timeframe}').publish('CRIT')
+	EOF
+
+	rule {
+		description           = "has not reported in ${var.heartbeat_timeframe}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "cluster_status" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ElasticSearch cluster status"
+
+	program_text = <<-EOF
+		A = data('ClusterStatus.red', filter=filter('namespace', 'AWS/ES') and filter('stat', 'upper') and ${module.filter-tags.filter_custom})${var.cluster_status_aggregation_function}.${var.cluster_status_transformation_function}(over='${var.cluster_status_transformation_window}')
+		B = data('ClusterStatus.yellow', filter=filter('namespace', 'AWS/ES') and filter('stat', 'upper') and ${module.filter-tags.filter_custom})${var.cluster_status_aggregation_function}.${var.cluster_status_transformation_function}(over='${var.cluster_status_transformation_window}')
+		detect(when(A >= ${var.cluster_status_threshold_critical})).publish('CRIT')
+		detect(when(B >= ${var.cluster_status_threshold_warning})).publish('WARN')
```

Comment:
I don't know if it stops to evaluate to the first positive "detect" function but if not we will have 2 notificaitons (warning + critical) if cluster status == cluster_status_threshold_critical

Reply:
we need to have only one alert : 
- warning if yellow
- critical if red
that is all

Reply:
Also I think for this detector specifically we can hardcode the values because it does not make any sens to allow changing thresholds